proxy/envoy/config.go

```diff
@@ -503,6 +503,19 @@ func buildInboundListeners(mesh *proxyconfig.ProxyMeshConfig, sidecar proxy.Node
 				Routes:  []*HTTPRoute{route},
 			}
 
+			// Websocket enabled routes need to have an explicit use_websocket : true
+			// This setting needs to be enabled on Envoys at both sender and receiver end
+			if protocol == model.ProtocolHTTP {
+				// get all the route rules applicable to the instances
+				rules := config.RouteRulesByDestination(instances)
```

Comment:
That is fine.. All we need is to know the URIs that support websockets and setup special route entries in envoy with websocket enabled. We don't have to do source based routing or any form of matching. This is just a roundabout way to collect websocket enabled URLs. I was hoping that we would put in some caching in that config object such that we dont have to compute this again and again.